From 115cb7ab7d3b87fe43b1fe9b05ec894b1fcfb5cf Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 7 Dec 2011 18:26:23 -0500
Subject: [PATCH] drm/edid: Try harder to fix up broken headers

There's no reason to force the first byte to be correct if we're already
scoring how correct the header is.

See also: https://bugzilla.redhat.com/show_bug.cgi?id=722909

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 drivers/gpu/drm/drm_edid.c |   18 ++++++++----------
 1 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/drm_edid.c b/drivers/gpu/drm/drm_edid.c
index 7425e5c..8b16a49 100644
--- a/drivers/gpu/drm/drm_edid.c
+++ b/drivers/gpu/drm/drm_edid.c
@@ -154,16 +154,14 @@ drm_edid_block_valid(u8 *raw_edid)
 	int i;
 	u8 csum = 0;
 	struct edid *edid = (struct edid *)raw_edid;
-
-	if (raw_edid[0] == 0x00) {
-		int score = drm_edid_header_is_valid(raw_edid);
-		if (score == 8) ;
-		else if (score >= 6) {
-			DRM_DEBUG("Fixing EDID header, your hardware may be failing\n");
-			memcpy(raw_edid, edid_header, sizeof(edid_header));
-		} else {
-			goto bad;
-		}
+	int score = drm_edid_header_is_valid(raw_edid);
+
+	if (score == 8) ;
+	else if (score >= 6) {
+		DRM_DEBUG("Fixing EDID header, your hardware may be failing\n");
+		memcpy(raw_edid, edid_header, sizeof(edid_header));
+	} else {
+		goto bad;
 	}
 
 	for (i = 0; i < EDID_LENGTH; i++)
-- 
1.7.7.3

